### PR TITLE
#609 New API: Reroute all flows which go through a particular ISL.

### PR DESCRIPTION
**Pull request by dpoltavets**
Closes: #609.

---
This change is [Reviewable](https://reviewable.io/reviews/telstra/open-kilda/1724)

### CODE REVIEW

File: services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/FlowOperationsBolt.java [hunk 1]
```diff
@@ -79,9 +88,31 @@ public void prepare(Map stormConf, TopologyContext context, OutputCollector coll
...
+        Integer srcPort = message.getSource().getPortNumber();
+        SwitchId dstSwitch = message.getDestination().getDatapath();
+        Integer dstPort = message.getDestination().getPortNumber();
+
+        List<String> flowIds = flowOperationsService.getFlowIdsForLink(srcSwitch, srcPort, dstSwitch, dstPort).stream()
+                .map(FlowPair::getForward)
+                .map(Flow::getFlowId)
+                .collect(Collectors.toList());
+
+        flowIds.forEach(flowId -> {
```
Comment (IlyaMoiseev) [inline services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/FlowOperationsBolt.java]: No needs to create collection with flowIds, you may process them with forEach on Stream

Comment (dpoltavets) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/FlowOperationsBolt.java]: This collection is a return value.

File: docs/design/usecase/link-reroute-all-flows-which-go-through.puml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+Neo4j -> NbWorkerTopology : Isl
+deactivate Neo4j
+
+alt
+
+NbWorkerTopology -> Neo4j : Query to get flows for\na particular link
+activate Neo4j
+Neo4j -> NbWorkerTopology : Flows
+deactivate Neo4j
+NbWorkerTopology -> kilda.northbound : InfoData\nFlowsResponse
```
[the remark this answers is not in the document]
Comment (dpoltavets) [reply docs/design/usecase/link-reroute-all-flows-which-go-through.puml]: My mistake. Fixed.

**Comment by dpoltavets** [comment]
___
*[services/wfm/src/main/java/org/openkilda/wfm/topology/flow/bolts/SplitterBolt.java, line 88 at r1](https://reviewable.io/reviews/telstra/open-kilda/1724#-LUEqybvBgFBM5UTJG9m:-LUU4bR--DDfXnlM9BwG:b-hwuk3f) ([raw file](https://github.com/telstra/open-kilda/blob/e0454493c766d93b4f760549a281345d0b32d791/services/wfm/src/main/java/org/openkilda/wfm/topology/flow/bolts/SplitterBolt.java#L88)):*
<details><summary><i>Previously, timofei-durakov (Timofey Durakov) wrote…</i></summary><blockquote>

Is it safe to add this condition? Don't we have cases where destination is empty?
</blockquote></details>

Yes, we can remove it. It was tested and no problems were found.

**Pull request merged**